service/controller/resource/appfinalizer/delete.go

```diff
@@ -24,19 +26,21 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 	// We keep the finalizer for the app-operator app CR so the resources in
 	// the management cluster are deleted.
 	o := metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("%s!=%s", label.AppKubernetesName, "app-operator"),
+		LabelSelector: fmt.Sprintf("%s=%s,%s!=%s", label.Cluster, key.ClusterID(&cr), label.AppKubernetesName, "app-operator"),
```

Comment:
this seems like a change of logic. Before you were looking at all apps in a namespace, now you change the label selector. Is this intended?

Reply:
we ran g8sclient with injecting key.ClusterID(&cr)
by now we only add it into the label selector because there's no other way but it's the same logic

Reply:
awesome